add sort direction to Global search

adamdaventryGov opened the pull request:
https://tracker.nci.nih.gov/browse/CTDC-1617 to fix this issue I had to add sort direction to bento.

adamdaventryGov committed:
add sort direction to Global search

to fix https://tracker.nci.nih.gov/browse/CTDC-1617

AustinSMueller commented on model/search/filter/GlobalQueryFilter.java:
```diff
@@ -36,7 +36,15 @@ public SearchSourceBuilder getSourceFilter() {
                 );
         // Set Sort
-        if (query.getFilter().getDefaultSortField() !=null) builder.sort(query.getFilter().getDefaultSortField(), SortOrder.DESC);
+        if ("ASC".equalsIgnoreCase(query.getFilter().getSortDirection())){
```
functionally this looks ok but it can be simplified and there is a lot of duplicated calls here. I would recommend storing the results of query.getFilter().getDefaultSortField() and query.getFilter().getSortDirection() in variables before the if statements. The logic here can also be simplified as shown below:

```
if default sort field is not null
    if sort direction is ASC
        sort ascending
    else
        sort descending
```

knockknockyoo commented on model/search/filter/GlobalQueryFilter.java:
```diff
@@ -36,7 +36,15 @@ public SearchSourceBuilder getSourceFilter() {
+            if (query.getFilter().getDefaultSortField() !=null) builder.sort(query.getFilter().getDefaultSortField(), SortOrder.DESC);
+        }
+        else{
+            if (query.getFilter().getDefaultSortField() !=null) builder.sort(query.getFilter().getDefaultSortField(), SortOrder.DESC);
```
if (query.getFilter().getDefaultSortField() != null) {
    String sortDirection = query.getFilter().getSortDirection();
    SortOrder order = "ASC".equalsIgnoreCase(sortDirection) ? SortOrder.ASC : SortOrder.DESC;
    builder.sort(sortField, order);
}
I think this could be more straightforward. Also, double check with Ming or Austin to confirm which branch to use before merging.

adamdaventryGov committed:
simplified code

as suggested , added a string to hold the sort direction value and reduced if statements

adamdaventryGov committed:
simplified the logic

knockknockyoo reviewed (approved):
Perfect!

adamdaventryGov committed:
further simplification of the code

further simplification of the code
puts remaining if statements in one line and defaults to DESC if result is not ASC

knockknockyoo commented on model/search/filter/GlobalQueryFilter.java:
```diff
@@ -36,8 +36,17 @@ public SearchSourceBuilder getSourceFilter() {
+      
+
+        String defaultSortField = query.getFilter().getDefaultSortField();
+        if (query.getFilter().getDefaultSortField() !=null){
```
please take a look.
if (query.getFilter().getDefaultSortField() !=null){ => if (defaultSortField!=null){

adamdaventryGov committed:
used string to compare instead of function call

used string to compare instead of function call
